piecrust/src/vm.rs

```diff
@@ -41,11 +114,14 @@ impl VM {
         let session_commits = SessionCommits::from(
             base_memory_path.join(SESSION_COMMITS_FILENAME),
         )?;
+        let modules = read_modules(&base_memory_path)?;
```

Comment:
I am a bit discomforted that VM's `new()` is doing actual directory scanning. Could we make an explicit step for scanning so that we have more control over it, IMO new() should be a lightweight function without performing IO.

Reply:
It seems to me that it doesn't make sense to instantiate a `VM` without knowledge of the modules it has deployed. Are you suggesting the instantiation process should look something like this: ?

```rust
let vm = VM::new(path)?;
vm.load()?;
```

This seems cumbersome and I can't see a usecase for just using `VM::new` then.

Reply:
I'd name it sth different than `new` then, otherwise I have no objections